org.adempiere.base/src/org/compiere/model/MInvoice.java

```diff
@@ -3347,4 +3349,65 @@ public void updateFrom(MOrder order)
 			}
 		}
 	}
+	
+	/**
+	 * Allocate Invoice for Completed Unallocated Pre-payments
+	 */
+	private boolean allocateFromOrder() {
+		// Get Set Of OrderIds from Lines
+		HashSet<Integer> orderIDSet = new HashSet<>();
+
+		if(getC_Order_ID() > 0)
+			orderIDSet.add(getC_Order_ID());
+
+		MInvoiceLine[] lines = getLines();
+		for(MInvoiceLine line : lines) {
+			if(line.getC_OrderLine_ID() <= 0)
+				continue;
+
+			 I_C_OrderLine orderLine = line.getC_OrderLine();
+			 if(orderLine == null)
+				 continue;
+			 if (line.getC_OrderLine_ID() > 0)
+				 orderIDSet.add(orderLine.getC_Order_ID());
+		}
+
+		if(orderIDSet.isEmpty())
+			return false;
+
+		StringBuilder whereClause = new StringBuilder("docstatus = 'CO' AND NOT EXISTS( SELECT 1 FROM c_allocationline al where al.c_payment_id=c_payment.c_payment_id) AND IsAllocated='N' AND COALESCE(C_Invoice_ID, 0)=0");

```

Comment:
you can't save 0 into C_Invoice_ID, so it should be just `C_Invoice_ID IS NULL`

Reply:
Changed to `AND C_Invoice_ID IS NULL`